contracts/YearnV2Strat.sol

```diff
@@ -0,0 +1,145 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IYToken.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+contract YTokenStrat is IStrat {
+    
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IVault public vault;
+    IYToken public yToken;
+    IERC20 public underlying;
+    Timelock public timelock;
+    bool public depositsPaused; // pausing only applies to Yearn deposits. Not strategy
+    uint public withdrawalCap; // 0 implies no cap
+    uint public unutilized; // similar to Uniswap reserves. Discrepancies are dangerous
+    string public name = "Yearn V2"; // for display purposes only
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == vault.owner()); // vault owner is strat owner
+        _;
+    }
+
+    constructor(IVault vault_, IYToken yToken_) {
+        vault = vault_;
+        yToken = yToken_;
+        timelock = Timelock(vault.timelock()); // use the same timelock from the vault
+        underlying = IERC20(yToken_.token());
+        underlying.safeApprove(address(yToken), uint(-1)); // intentional underflow
+    }
+
+    function invest() external override onlyVault {
+        uint amount = underlying.balanceOf(address(this)).sub(unutilized); // unutilized reserves do not count for incoming investment
+        require(amount > 0, "Nothing to invest");
+        if(!depositsPaused) {
+            uint max = yToken.availableDepositLimit();
+            yToken.deposit(Math.min(amount, max), address(this)); // respect Yearn's limits and leave the remainder here
+            if(max < amount) { // some funds are left unutilized
+                syncUntilized();
+            }
+        } else {
+            syncUntilized(); // just keep the entire amount here
+        }
+    }
+
+    function divest(uint amount) external override onlyVault {
+        require(amount > 0, "Nothing to divest"); // this fails when migrating away but there are no deposits. But in that case, what are we even migrating?
+        bool zeroUnutilized = unutilized == 0; // to save on SSTOREs
+        if(unutilized < amount) {
+            uint missingAmount = amount.sub(unutilized);
+            if(withdrawalCap > 0) { // 0 implies no cap
+                require(missingAmount <= withdrawalCap, "Reached withdrawal cap"); // Big withdrawals can cause slippage on Yearn's side. Users must split into multiple txs
+            }
+            yToken.withdraw(sharesForAmount(missingAmount) + 1, address(this));  // +1 is a fix for a rounding issue
```

Comment:
you have removed the check?

I'd do a balance before and balance after check to make sure the difference is within dust threshold

Reply:
Any dust will be added to the unutilized balance after and will be added to the next withdrawal.
The check is dangerous because we cannot let the function fail.